From fecaa0c74f66643d5bd55422b426aefeaaaa25f2 Mon Sep 17 00:00:00 2001
From: Kalesh AP <kalesh-anakkur.purayil@broadcom.com>
Date: Mon, 17 Nov 2025 11:43:06 +0530
Subject: [PATCH 1/1] RDMA/bnxt_re: Fix wrong check for CQ coalesc support
Git-commit: fecaa0c74f66643d5bd55422b426aefeaaaa25f2
Patch-mainline: v6.19-rc1
References: jsc#PED-15349

Driver is not creating the debugfs hooks for CQ coalesc parameters
because of a wrong check. Fixed the condition check inside
bnxt_re_init_cq_coal_debugfs().

Fixes: cf2749079011 ("RDMA/bnxt_re: Add a debugfs entry for CQE coalescing tuning")
Signed-off-by: Kalesh AP <kalesh-anakkur.purayil@broadcom.com>
Link: https://patch.msgid.link/20251117061306.1140588-1-kalesh-anakkur.purayil@broadcom.com
Signed-off-by: Leon Romanovsky <leon@kernel.org>
Acked-by: Nicolas Morey <nmorey@suse.com>
---
 drivers/infiniband/hw/bnxt_re/debugfs.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/infiniband/hw/bnxt_re/debugfs.c b/drivers/infiniband/hw/bnxt_re/debugfs.c
index d03f5bb0d890..88817c86ae24 100644
--- a/drivers/infiniband/hw/bnxt_re/debugfs.c
+++ b/drivers/infiniband/hw/bnxt_re/debugfs.c
@@ -454,7 +454,7 @@ static void bnxt_re_init_cq_coal_debugfs(struct bnxt_re_dev *rdev)
 	struct bnxt_re_dbg_cq_coal_params *dbg_cq_coal_params;
 	int i;
 
-	if (_is_cq_coalescing_supported(rdev->dev_attr->dev_cap_flags2))
+	if (!_is_cq_coalescing_supported(rdev->dev_attr->dev_cap_flags2))
 		return;
 
 	dbg_cq_coal_params = kzalloc(sizeof(*dbg_cq_coal_params), GFP_KERNEL);
-- 
2.52.0

